src/main/scala/firrtl/Emitter.scala

```diff
@@ -751,13 +754,23 @@ class VerilogEmitter extends SeqTransform with Emitter {
     passes.VerilogPrep)
 
   def emit(state: CircuitState, writer: Writer): Unit = {
-    writer.write(preamble)
 
     val circuit = runTransforms(state).circuit
     val moduleMap = circuit.modules.map(m => m.name -> m).toMap
-    circuit.modules.foreach {
-      case m: Module => emit_verilog(m, moduleMap)(writer)
-      case _: ExtModule => // do nothing
+    val (hasRANDOMIZE, emissions) = circuit.modules.foldLeft((false, Seq[String]())) { (trackers, m) =>
+      (trackers, m) match {
+        case ((allRANDOMIZE, seq), m: Module) =>
+          val (emission, rands) = emit_verilog(m, moduleMap)
+          (allRANDOMIZE || rands, seq ++ Seq(emission))
+        case ((allRANDOMIZE, seq), _: ExtModule) => (allRANDOMIZE, seq)
+        case _ => sys.error("Shouldn't be here")
```

Comment:
I believe we want to use the `error` method defined in Utils rather than `sys.error` right?